archlinuxcn/llvm-git/PKGBUILD

```diff
@@ -0,0 +1,404 @@
+#!/usr/bin/bash
+
+
+# Based on LordHeavy's LLVM
+pkgbase=llvm-git
+pkgname=('lldb-git' 'lld-git' 'polly-git' 'compiler-rt-git' 'clang-git' 'llvm-ocaml-git' 'llvm-libs-git' 'llvm-git'  'lib32-llvm-libs-git' 'lib32-clang-git' 'lib32-llvm-git') 
+pkgdesc='Low Level Virtual Machine (git version)'
+pkgver=12.0.0_r376876.6f0f0220380f
+pkgrel=1
+groups=('chaotic-mesa-git')
+arch=('x86_64')
+url="https://llvm.org/"
+license=('custom:Apache 2.0 with LLVM Exception')
+makedepends=('git' 'cmake' 'ninja' 'libffi' 'libedit' 'ncurses' 'libxml2'
+           'python-sphinx' 'ocaml' 'ocaml-ctypes' 'ocaml-findlib' 
+            'python-sphinx' 'python-recommonmark' 'cuda' 'ocl-icd' 'opencl-headers'
+             'swig' 'python' 'lib32-gcc-libs' 'lib32-libffi' 'libunwind' 'lib32-libunwind'
+               'lib32-libxml2' 'lib32-zlib' 'python2' ) #tensorflow
+
+source=("llvm-project::git+https://github.com/llvm/llvm-project.git"
+        "llvm-config.h")
+
+md5sums=('SKIP'
+         '295c343dcd457dc534662f011d7cff1a')
+sha512sums=('SKIP'
+            '75e743dea28b280943b3cc7f8bbb871b57d110a7f2b9da2e6845c1c36bf170dd883fca54e463f5f49e0c3effe07fbd0db0f8cf5a12a2469d3f792af21a73fcdd')
+options=('staticlibs')
+
+# NINJAFLAGS is an env var used to pass commandline options to ninja
+# NOTE: It's your responbility to validate the value of $NINJAFLAGS. If unsure, don't set it.
+
+_python2_optimize() {
+  python2 -m compileall "$@"
+  python2 -O -m compileall "$@"
+}
+
+_python3_optimize() {
+  python -m compileall "$@"
+  python -O -m compileall "$@"
+  python -OO -m compileall "$@"
+}
+
+pkgver() {
+    cd llvm-project/llvm
+
+    # This will almost match the output of `llvm-config --version` when the
+    # LLVM_APPEND_VC_REV cmake flag is turned on. The only difference is
+    # dash being replaced with underscore because of Pacman requirements.
+    local _pkgver=$(awk -F 'MAJOR |MINOR |PATCH |)' \
+            'BEGIN { ORS="." ; i=0 } \
+             /set\(LLVM_VERSION_/ { print $2 ; i++ ; if (i==2) ORS="" } \
+             END { print "\n" }' \
+             CMakeLists.txt)_r$(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+    echo "$_pkgver"
+
+}
+
+prepare() {
+ 
+    cd llvm-project 
+    # llvm-project contains a lot of stuff, remove parts that aren't used by this package
+    rm -rf debuginfo-tests libclc libcxx libcxxabi llgo openmp parallel-libs pstl libc
+
+    rm -rf "$srcdir"/fakeinstall*
+    
+    export _ocamlver=$(ocamlc --version)
+    export _pythonver=$(python -V | sed 's/.* \([0-9]\).\([0-9]\).*/\1.\2/')
+}
+
+build() {
+    export CFLAGS="$(echo "$CFLAGS" | sed -e "s/-fstack-protector-strong//") -I/usr/include/tensorflow "
```

Comment:
And just curious, did the stack protector break something or was this just to get better performance?

I'm also pretty sure that it'll be more reliable if you add a `-fno-stack-protector` before/after instead. Had arch changes the default protection option this version will most likely break due to string mismatch.